### ISSUE TITLE
Questions about code（tensorflow2）operational efficiency

### ISSUE THREAD

**Issue by liyuanfeng747** [knowledgedefinednetworking/DRL-GNN#14]
Dear Authors, Recently I have been studying your papers and code. I would like to extend your work to multi-agent. After some more attempts, I have made some progress in my work, however I have encountered the problem that the gradient of the algorithm is too slow to be updated. In my current scenario, there are a total of 8 independent graph neural networks that need to be updated and trained. Even though I have used gpu for training, it still takes about 30 seconds to perform each round of updates. Since I don't know much about the working mechanism of tensorflow. I would like to ask what is causing this problem and is there any chance to improve it?

**Comment by paulalmasan** [comment]
Hi @liyuanfeng747 , having 8 GNNs sounds like many parameters to train, so it would explain why the gradient computation takes so much time. Maybe you could try reducing the number of GNNs or making them smaller. Also as a first approach I recommend to run the experiments on very small graphs

**Comment by liyuanfeng747** [comment]
dear author! i'm so glad to get your reply!&nbsp; Do you mean i should train in small graphs and reduce the state_dims to its link number? thank you very much!


s1mple
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "knowledgedefinednetworking/DRL-GNN"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月22日(星期二) 晚上9:39
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [knowledgedefinednetworking/DRL-GNN] Questions about code（tensorflow2）operational efficiency (Issue #14)





 
Hi @liyuanfeng747 , having 8 GNNs sounds like many parameters to train, so it would explain why the gradient computation takes so much time. Maybe you could try reducing the number of GNNs or making them smaller. Also as a first approach I recommend to run the experiments on very small graphs

**Comment by paulalmasan** [comment]
Both things you mention should help. If graphs are very small you can also reduce the parameter 'T'.

**Comment by liyuanfeng747** [comment]
> 

thank you very much , i should try these precious advices
